### DISCUSSION TITLE
crazyflie 2.1 > custom firmware > xyz.bin generating out of custom .config

### DISCUSSION THREAD

**Discussion by boomer319** [bitcraze/discussions#936] [Q&A] [answered]
Hello,

I am currently working with the crazyflie 2.1.

I followed the tutorial about the wifi video streamer (https://www.bitcraze.io/documentation/repository/aideck-gap8-examples/master/simple-examples/wifi-streamer/).

In the tutorials included in the wifi streamer tutorials you jump from generating and editing the .config with kbuild straight to flashing (https://www.bitcraze.io/documentation/repository/crazyflie-firmware/master/building-and-flashing/build/).

I made a custom .config file. However I don't see how I am supposed to make a flashable xyz.bin file from the custom .config.

I am running Ubuntu 22.04. If that's important.

How do I make a flashable xyz.bin file to flash to the crazyflie 2.1 based on my custom .config?

xyz in xyz.bin is a placeholder for whatever you might name the custom firmware.

**Comment by knmcguire** [comment, marked as answer]
Hi!

So what you are doing, is to make a custom crazyflie firmware file that has the wifi credentials info. With startup, the crazyflie will communicate that info to the ESP chip on the aideck and create an access point/connect to an accesspoint what you have entered in the custom kbuild config by the menu gui.

So if you have changed the config file with the [kbuild instructions](https://www.bitcraze.io/documentation/repository/crazyflie-firmware/master/development/kbuild/). You should build the firmware again with

`make -j 12`

This will generate cf2.bin in the build folder automatically. Don't do `make cf2_defconfig` as that will overwrite your custom config file. 

Then you can flash the new crazyflie firmware via these instructions https://www.bitcraze.io/documentation/repository/crazyflie-firmware/master/building-and-flashing/build/#flashing. This will flash the generated `cf2.bin` in the build folder.

Do mind, that for this example to work (the aideck taking the credentials from the crazyflie firmware), you will need to reflash your aideck as well via [these instructions](https://www.bitcraze.io/documentation/repository/aideck-gap8-examples/master/simple-examples/wifi-streamer/#building-and-flashing-the-example). The wifi example that you had to flash with the [getting started with aideck tutorial](https://www.bitcraze.io/documentation/repository/aideck-gap8-examples/master/simple-examples/wifi-streamer/#building-and-flashing-the-example) is 'deaf' to the crazyflie firmware's instructions as it is meant to just test out your aideck as quickly as possible. So that is why it is important to reflash the aideck gap8 firmware with the new example to make it listen to the crazyflie autopilot. 

Does that make sense to you? let me know if we need to explain anything more.

**Reply by boomer319** [reply]
This does make sense. Thank you!

**Reply by knmcguire** [reply]
awesome! Keep us up to date on your progress and if you encounter another issue, please start a new discussion.
